File: esbmc.py/Testes02/validador_titulo/validador_somente_letras_titulo.cpp
#include <cctype>

#define MAX_LEN_TITLE 21

using namespace std;

bool validate_only_letters(const char title[MAX_LEN_TITLE], int len) {
    for (int i = 0; i < len; i++) {
        __ESBMC_assert(isalpha(title[i]), "O título deve conter apenas letras.");
    }
    return true;
}

int main() {
    char title[MAX_LEN_TITLE];
    int len;

    __ESBMC_assume(len >= 1 && len <= 20);

    validate_only_letters(title, len);

    return 0;
}

File: esbmc.py/Testes02/validador_titulo/validador_somente_letras_titulo.py
MAX_LEN_TITLE: int = 21

def validate_only_letters(title: str, length: int) -> bool:
    i: int = 0
    while i < length:
        assert title[i].isalpha(), "O título deve conter apenas letras."
        i += 1
    return True


def main() -> None:
    title: str = input()      
    length: int = len(title)

    assert 1 <= length <= 20  

    validate_only_letters(title, length)


main()
